bot/bot.py

```diff
@@ -101,6 +114,10 @@ def _recreate(self) -> None:
         self.http_session = aiohttp.ClientSession(connector=self._connector)
         self.api_client.recreate(force=True, connector=self._connector)
 
+    async def on_ready(self) -> None:
+        """Construct an asynchronous transport for the statsd client."""
+        await self.stats.create_socket()
```

Comment:
Did you intentional put it in an event that may be called on bot reconnect? If not, this should go in `_recreate()`.

Reply:
Right, but recreate is not an async function, would I have to use `create_task` again on the `loop` attribute of the bot class?

Reply:
It isn't critical, I can have the `create_socket` method check if there is an existing transport and close that before creating another one, but I'm not sure how to do it from `_recreate()`.

Reply:
Hmm right. It could go in `login`. If not, at the least the previous socket should be closed before you overwrite it.

Reply:
Not being able to have it in `_recreate` kind of makes the `clear()` function incomplete, but in practice we don't use `clear()`.

Reply:
Moved to login